crates/orchestrator/src/config.rs

```diff
@@ -90,37 +99,28 @@ pub async fn get_aws_config(settings_provider: &impl Settings) -> SdkConfig {
 pub async fn init_config() -> Arc<Config> {
     dotenv().ok();
 
+    let settings_provider = EnvSettingsProvider {};
+    let provider_config = ProviderConfig::AWS(Arc::new(get_aws_config(&settings_provider).await));
+
     // init starknet client
     let provider = JsonRpcClient::new(HttpTransport::new(
-        Url::parse(get_env_var_or_panic("MADARA_RPC_URL").as_str()).expect("Failed to parse URL"),
+        Url::parse(settings_provider.get_settings_or_panic("MADARA_RPC_URL").as_str()).expect("Failed to parse URL"),
     ));
 
-    let settings_provider = EnvSettingsProvider {};
-    let aws_config = Arc::new(get_aws_config(&settings_provider).await);
-
     // init database
     let database = build_database_client(&settings_provider).await;
     let da_client = build_da_client(&settings_provider).await;
     let settlement_client = build_settlement_client(&settings_provider).await;
     let prover_client = build_prover_service(&settings_provider);
-    let storage_client = build_storage_client(&settings_provider, ProviderConfig::AWS(Arc::clone(&aws_config))).await;
-    let alerts_client = build_alert_client(&settings_provider, ProviderConfig::AWS(Arc::clone(&aws_config))).await;
+    let storage_client = build_storage_client(&settings_provider, provider_config.clone()).await;
```

Comment:
can we not pass a reference

Reply:
Done, https://github.com/madara-alliance/madara-orchestrator/pull/107/commits/2b0fb3c53125ea96a0bed03d61d726cf9c6f1714